manager/orchestrator/task_reaper.go

```diff
@@ -75,13 +79,30 @@ func (tr *TaskReaper) Run() {
 				}] = struct{}{}
 				if len(tr.dirty) > maxDirty {
 					tr.tick()
+					if timer != nil {
+						timer.Stop()
+					}
+					timerCh = nil
```

Comment:
[timer.Stop()](https://golang.org/pkg/time/#Timer.Stop) `Stop does not close the channel, to prevent a read from the channel succeeding incorrectly.`

I feel like this handling between timer and timeCh can be simplified. You can create the timer and channel outside `for` loop. Inside `for` loop, just call `timer.Stop` or `timer.Reset`. No need to check `if timer == nil` or update `timerCh`.